From fc4cfa79cf0389b9ed3dcbfdfab092126fbe9406 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Wed, 2 Oct 2024 15:45:54 +0200
Subject: [PATCH] dts: arm64: st: add wakeup source for Ethernets for stm32mp23

When the "wakeup-source" flag is found, Ethernet marks
the device as wakeup capable and enables the wakeup by
default. (with WoL event)

Change-Id: I9d3a843935f5abf7dd900a5f9aa3e0e47129406d
Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/408531
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp231.dtsi | 7 +++++--
 arch/arm64/boot/dts/st/stm32mp233.dtsi | 7 +++++--
 2 files changed, 10 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -2012,8 +2012,10 @@
 				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 				reg = <0x482c0000 0x4000>;
 				reg-names = "stmmaceth";
-				interrupts-extended = <&intc GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>;
-				interrupt-names = "macirq";
+				interrupts-extended = <&intc GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>,
+						      <&exti1 68 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "macirq",
+						  "eth_wake_irq";
 				clock-names = "stmmaceth",
 					      "mac-clk-tx",
 					      "mac-clk-rx",
@@ -2033,6 +2035,7 @@
 				snps,tso;
 				access-controllers = <&rifsc 60>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 				snps,mtl-rx-config = <&mtl_rx_setup_1>;
 				snps,mtl-tx-config = <&mtl_tx_setup_1>;
--- a/arch/arm64/boot/dts/st/stm32mp233.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp233.dtsi
@@ -119,8 +119,10 @@
 		compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 		reg = <0x482d0000 0x4000>;
 		reg-names = "stmmaceth";
-		interrupts-extended = <&intc GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "macirq";
+		interrupts-extended = <&intc GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>,
+				      <&exti1 70 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "macirq",
+				  "eth_wake_irq";
 		clock-names = "stmmaceth",
 			      "mac-clk-tx",
 			      "mac-clk-rx",
@@ -140,6 +142,7 @@
 		snps,tso;
 		access-controllers = <&rifsc 61>;
 		power-domains = <&CLUSTER_PD>;
+		wakeup-source;
 		status = "disabled";
 		snps,mtl-rx-config = <&mtl_rx_setup_2>;
 		snps,mtl-tx-config = <&mtl_tx_setup_2>;
